File: Java/PuntoVentaDulces/src/Clases/Producto.java
package Clases;

public abstract class Producto {
    protected String codigo;
    protected String nombre;
    protected double precio;
    protected int stock;

    public Producto(String codigo, String nombre, double precio, int stock) {
        this.codigo = codigo;
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
    }

    public String getCodigo() { return codigo; }
    public String getNombre() { return nombre; }
    public double getPrecio() { return precio; }
    public int getStock() { return stock; }
    
    public void setPrecio(double precio) { this.precio = precio; }
    public void setStock(int stock) { this.stock = stock; }

    @Override
    public String toString() {
        return String.format("[%s] %-15s | Precio: $%.2f | Stock: %d", codigo, nombre, precio, stock);
    }
}

File: PuntoVentaDulces/PuntoVentaDulces/Producto.cs
namespace Clases
{
    public abstract class Producto
    {
        protected string codigo;
        protected string nombre;
        protected double precio;
        protected int stock;

        public Producto(string codigo, string nombre, double precio, int stock)
        {
            this.codigo = codigo;
            this.nombre = nombre;
            this.precio = precio;
            this.stock = stock;
        }
        public string GetCodigo() { return codigo; }
        public string GetNombre() { return nombre; }
        public double GetPrecio() { return precio; }
        public int GetStock() { return stock; }

        public void SetPrecio(double precio) { this.precio = precio; }
        public void SetStock(int stock) { this.stock = stock; }

        public override string ToString()
        {

            return $"[{codigo}] {nombre,-15} | Precio: ${precio:0.00} | Stock: {stock}";
        }
    }
}